packages/slate-react/test/index.js

```diff
@@ -1 +1,46 @@
-describe('slate-react', () => {})
+import * as Slate from 'slate'
+import * as SlateReact from '..'
+import { JSDOM } from 'jsdom'
+import React from 'react'
+import TestRenderer from 'react-test-renderer'
+import assert from 'assert'
+
+describe('slate-react', () => {
+  describe('Editable', () => {
+    describe('decorate', () => {
+      // stub out some DOM stuff to avoid crashes
+      before(() => {
+        const jsdom = new JSDOM()
+        global.window = jsdom.window
+        global.document = jsdom.window.document
+        global.Document = document.constructor
```

Comment:
I'm not too familiar with jsdom testing, but is there any way we could extract this global setting from this one suite, so it's done for all suites (when they exist)? Or must it be done a per-suite basis? 

In fact, could you put the whole `decorate` test in a `decorate.spec.ts` or similar file? The `index` could just be common utilities among tests for example.

Reply:
I mostly use Jest which has special support for jsdom https://jestjs.io/docs/configuration#testenvironment-string. I didn't find an equivalent for Mocha but there is https://github.com/rstacruz/jsdom-global which does pretty much what I do here.

I think it could be done for the whole test file, but I didn't want to presume. I figured someone with more Mocha experience would have opinions about how to set things up.

Reply:
I would suggest we land this as it's an improvement all around, and then we can do a separate PR to look at extracting into a global setting.